task/attest/src/main.rs

```diff
@@ -268,6 +284,102 @@ impl idl::InOrderAttestImpl for AttestServer {
 
         Ok(len)
     }
+
+    fn attest(
+        &mut self,
+        _: &userlib::RecvMessage,
+        nonce: Leased<R, [u8]>,
+        dest: Leased<W, [u8]>,
+    ) -> Result<(), RequestError<AttestError>> {
+        if nonce.len() != Self::NONCE_SIZE {
+            let err = AttestError::BadLease;
+            ringbuf_entry!(Trace::AttestError(err));
+            return Err(err.into());
+        }
```

Comment:
If this is a fixed size it might be worth enforcing this via `LenLimit` (which is an upper bound but might catch some errors at build time)

Reply:
I'm thinking we'll probably want to be able to support larger nonces in the future. The current min / max are both 32 bytes but w/ `LenLimit` I can probably set a range. Not sure what a good max value would be (hard to predict the future). The `sprot` plumbing can tolerate up to a 512 byte lease in both directions so that's not the limiting factor. The impact of allocating memory to hold the nonce in the attest task is likely to be the most significant trade off. A range like `32..128` is probably "future proof" enough and won't push memory usage too far. Will work something up before flipping the `draft` label off.

Reply:
I've updated this PR such that the `attest` idol op in both `idl/attest.idol` and `idl/sprot.idol` have an upper bound of `128` bytes. The lower bound of `32` bytes is enforced with a conditional in the relevant functions. There's really no reason we couldn't accept a `nonce` that's `512` bytes. But there's little reason to use a nonce larger than `32` bytes so we impose the `128` byte upper bound to keep memory usage down. In the future, increasing this size would have a minimal impact. The important thing is to prevent the caller from using a nonce that's less than `32` bytes.